### PR TITLE
Add filter fields and pagination

### PR DESCRIPTION
**Pull request by ricardosantacattarina**

### CODE REVIEW

**Comment by ricardosantacattarina** [comment]
[[Docs] Include endpoints with filter](https://trello.com/c/D1iZY2Ya/454-docs-include-endpoints-with-filter)

File: docs/beta_machines_endpoints.md [hunk 1]
```diff
@@ -43,12 +43,22 @@ Description | Endpoints
...
-Get the list of machines that are related to a given LeafUserId. 
+Get the list of machines that are related to a given LeafUserId. Some fields can be sent as query parameters to filter the machines, such as:
+- `name`, to filter by machine name
+- `provider`, to fitler by Provider
+- `providerOrganizationId`, to filter by the organization id
+- `serialNumber`, to filter by the machine serial number
+- `originType`, to filter by the origin of the machine, must be either USER_CREATED, FILE_POOLED or PROVIDER_POOLED
+- `createdAt`, to filter by the exact time of the record creation
+- `beforeCreatedAt`, filters for records created before the datetime
+- `afterCreatedAt`, filters for records created after the datetime
```
Comment (fabyan-alexander) [inline docs/beta_machines_endpoints.md:54]: Should we repeat the text "to filter by"?
Maybe we could pass the supported value, like **text/uuid/iso8601**, what do you think?

File: docs/beta_machines_endpoints.md [hunk 2]
```diff
@@ -281,8 +291,22 @@ A machine record with more details.
...
-Get the details of machine operations files given a machine id.
+Get the details of machine operations files given a machine id. Some fields can be sent as query parameters to filter the machine files, such as:
+
+- `leafFileId`, to filter by the leafFile ID
+- `startTime`, to filter by the exact time of the start of the operation
+- `endTime`, to filter by the exact time of the end of the operation
+- `originType`, to filter by the origin of the machine file, must be either USER_CREATED, FILE_POOLED or PROVIDER_POOLED
+- `createdAt`, to filter by the exact time of the record creation
+- `beforeCreatedAt`, filters for records created before the datetime
+- `afterCreatedAt`, filters for records created after the datetime
```
Comment (fabyan-alexander) [inline docs/beta_machines_endpoints.md:302]: Should we repeat the text "to filter by"?
Maybe we could pass the supported value, like **text/uuid/iso8601**, what do you think?

**Review by fabyan-alexander — CHANGES_REQUESTED** [review]
Nice update!
I think we could improve the description =]

File: docs/beta_machines_endpoints.md (hunk 2 above)
Comment (ricardosantacattarina) [reply docs/beta_machines_endpoints.md]: https://github.com/Leaf-Agriculture/docs/pull/197/commits/aba8e341ecbced2a5092fa0e18c0c1a88fe1de73 what about now?

File: docs/beta_machines_endpoints.md (hunk 1 above)
Comment (ricardosantacattarina) [reply docs/beta_machines_endpoints.md]: https://github.com/Leaf-Agriculture/docs/pull/197/commits/aba8e341ecbced2a5092fa0e18c0c1a88fe1de73 what about now?

Comment (fabyan-alexander) [reply docs/beta_machines_endpoints.md]: It is nice!

File: docs/beta_machines_endpoints.md (hunk 2 above)
Comment (fabyan-alexander) [reply docs/beta_machines_endpoints.md]: It is nice!

**Review by fabyan-alexander — APPROVED** [review]

**Pull request merged**